features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/modules/device.js

```diff
@@ -332,5 +334,75 @@ deviceModule = function () {
         return license;
     };
 
+    publicMethods.getOwnDevices = function () {
+        devicemgtService = devicemgtProps["httpsURL"] + "/common/device_manager";
+        listAllDevicesEndPoint = devicemgtService + "/device/user/" + user.username + "/all";
+        var result = get(listAllDevicesEndPoint, {}, "json");
+        var devices = result.data;
+        var device;
+        for (var d in devices){
+            device = devices[d];
+            device.assetId = publicMethods.getAssetId(device.deviceType);
+        }
+        return result;
+    };
+
+    publicMethods.getOwnDevicesCount = function () {
+        var carbonUser = session.get(constants.USER_SESSION_KEY);
+        devicemgtService = devicemgtProps["httpsURL"] + "/common/device_manager";
+        listAllDevicesEndPoint = devicemgtService + "/device/user/" + carbonUser.username + "/all/count";
+        return get(listAllDevicesEndPoint, {}, "json").data;
+    };
+
+    publicMethods.getUnGroupedDevices = function () {
+        var carbonUser = session.get(constants.USER_SESSION_KEY);
+        devicemgtService = devicemgtProps["httpsURL"] + "/common/device_manager";
+        listAllDevicesEndPoint = devicemgtService + "/device/user/" + carbonUser.username + "/ungrouped";
+        return get(listAllDevicesEndPoint, {}, "json").data;
+    };
+
+    publicMethods.getUnGroupedDevicesCount = function () {
+        var result = publicMethods.getUnGroupedDevices();
+        var devices = result.data;
+        var count = 0;
+        if (devices) {
+            count = devices.length;
+        }
+        result.data = count;
+        return result;
+    };
+
+    publicMethods.getAllPermittedDevices = function () {
+        var groupModule = require("../modules/group.js").groupModule;
+
+        var result = publicMethods.getUnGroupedDevices();
+        var unGroupedDevices = result.data;
+        var user_groups = groupModule.getGroups().data;
+        var allDevices = [];
+        var deviceCount = unGroupedDevices.length;
+        for (var g in user_groups) {
+            var deviceInGroup = user_groups[g].devices;
+            deviceCount += deviceInGroup.length;
+            if (deviceInGroup && deviceInGroup.length == 0) {
+                delete user_groups[g]["devices"];
+            }
+            var device;
+            for (var d in deviceInGroup){
```

Comment:
Acknowledge, Already fixed in moved code with commit https://github.com/wso2/carbon-device-mgt/commit/e537e2d841ca3c7946cd34e9da96f261e0ac94e1